src/System.Text.Json/tests/JsonArrayTests.cs

```
@@ -0,0 +1,344 @@
+            Assert.True(jsonArray.Contains(short.MaxValue));
+
+            jsonArray.Add(int.MaxValue);
+            Assert.Equal(int.MaxValue, ((JsonNumber)jsonArray[idx++]).GetInt32());
+            Assert.True(jsonArray.Contains(int.MaxValue));
+
+            jsonArray.Add(long.MaxValue);
+            Assert.Equal(long.MaxValue, ((JsonNumber)jsonArray[idx++]).GetInt64());
+            Assert.True(jsonArray.Contains(long.MaxValue));
+
+            jsonArray.Add(3.14f);
+            Assert.Equal(3.14f, ((JsonNumber)jsonArray[idx++]).GetSingle());
+
+            jsonArray.Add(3.14);
+            Assert.Equal(3.14, ((JsonNumber)jsonArray[idx++]).GetDouble());
+
+            jsonArray.Add(sbyte.MaxValue);
+            Assert.Equal(sbyte.MaxValue, ((JsonNumber)jsonArray[idx++]).GetSByte());
+
+            jsonArray.Add(ushort.MaxValue);
+            Assert.Equal(ushort.MaxValue, ((JsonNumber)jsonArray[idx++]).GetUInt16());
+
+            jsonArray.Add(uint.MaxValue);
+            Assert.Equal(uint.MaxValue, ((JsonNumber)jsonArray[idx++]).GetUInt32());
+
+            jsonArray.Add(ulong.MaxValue);
+            Assert.Equal(ulong.MaxValue, ((JsonNumber)jsonArray[idx++]).GetUInt64());
+
+            jsonArray.Add(decimal.One);
+            Assert.Equal(decimal.One, ((JsonNumber)jsonArray[idx++]).GetDecimal());
+        }
+
+        [Fact]
+        public static void TestInsert()
+        {
+            var jsonArray = new JsonArray();
+
+            jsonArray.Insert(0, "value");
+            Assert.Equal("value", (JsonString)jsonArray[0]);
+
+            jsonArray.Insert(0, true);
+            Assert.True(((JsonBoolean)jsonArray[0]).Value);
+
+            jsonArray.Insert(0, byte.MaxValue);
+            Assert.Equal(byte.MaxValue, ((JsonNumber)jsonArray[0]).GetByte());
+
+            jsonArray.Insert(0, short.MaxValue);
+            Assert.Equal(short.MaxValue, ((JsonNumber)jsonArray[0]).GetInt16());
+
+            jsonArray.Insert(0, int.MaxValue);
```

Negative tests are just as important as positive tests.  For each Add method there should be a test that inserting beyond Length or inserting negative numbers should throw an exception, and that exception should be validated (presumably ArgumentOutOfRangeException for a parameter named "index").